node.go

```diff
@@ -229,7 +241,13 @@ func (n *Node) recvLoop() {
 				}).Warnf("failed to parse packet: %v", err)
 				continue
 			}
-			go n.handlePacket(p)
+
+			// at this point we assume that p contains an
+			// unmarshalled packet that must have a valid
+			// opcode which we can now extract and handle
+			// the packet by calling the corresponding
+			// callback
+			go n.handlePacket(p, code.OpCode(binary.BigEndian.Uint16([]byte{payload.data[8], payload.data[9]})))
```

Comment:
On second thought, please also move the opcode extraction logic to the `handlePacket` function, just for clarity and to have all the parsing and callback in the same place.

Reply:
I've thought about that but decided against it to avoid passing the payload data into the `handlePacket()` so it remains in `recvLoop()`.

Reply:
Ah, that makes perfect sense.